opensearch-transport/lib/opensearch/transport/client.rb

```diff
@@ -330,14 +330,17 @@ def __parse_host(host)
       # @api private
       #
       def __auto_detect_adapter
+        # Get the Faraday adapter list without initializing it.
+        adapters = Faraday::Adapter.instance_variable_get(:@registered_middleware) || {}
+        # Pick an adapter that has both the client and adapter defined.
         case
-        when defined?(::Patron)
+        when defined?(::Patron) && adapters.key?(:patron)
```

Comment:
I check the keys with each condition so it will go on to detect the next HTTP client if the adapter isn't defined.